portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -71,105 +74,126 @@ public static int getSize(Map<String, LinkedList<BlogMessage>> map) {
   }
 
   // Returns a requested amount of all BlogMessages in map
-  public static LinkedList<BlogMessage> getMessages(Map<String, LinkedList<BlogMessage>> map, int limit) {
+  // Returned values contain the most recent messages.
+  public LinkedList<BlogMessage> getMessages(int limit) {
     if (limit < 0) {
-      System.out.println("Cannot load this amount.");
       return new LinkedList<>();
     }
 
-    LinkedList<BlogMessage> values = getMessages(map);
-    if (limit > values.size()) {
-      limit = values.size();
-    } 
-
-    System.out.println("Fetching requested amount of [LinkedList] Values for all tags...");
+    Set<String> keys = map.keySet();
+    LinkedList values = new LinkedList();
 
-    Iterator<BlogMessage> iterateList = values.iterator();
-    LinkedList<BlogMessage> limitedValues = new LinkedList<BlogMessage>();
-    int i = 0;
-    while (iterateList.hasNext() && i < limit){
-      limitedValues.addLast(iterateList.next());
-      i++;
+    for (String key : keys) {
+      if (limit <= 0) {break;}
+      Iterator iterator;
+      int tagMessagesSize = (map.get(key)).size(); 
+
+      // startPoint is from (n0.of.messages in each tag - limit) if the list contains more messages
+      // than the limit,
+      // That way, returned values contain the most recent messages.
+      if (tagMessagesSize > limit) {
+        int startPoint = tagMessagesSize - limit;
```

Comment:
"startPoint" is a local variable that is used only once right next line. Would be good to just pass "tagMassageSize - limit" to listIterator() next line.

Reply:
Same applies else where.